app/controllers/oligrapher_controller.rb

```diff
@@ -174,6 +174,32 @@ def get_edges
     render json: edges
   end
 
+  def get_interlocks
+    return head :bad_request if params[:entity1_id].blank?
+    return head :bad_request if params[:entity2_id].blank?
+    return head :bad_request if params[:entity_ids].blank?
+
+    num = params.fetch(:num, 10).to_i
+    interlock_ids = Entity.interlock_ids(params[:entity1_id], params[:entity2_id])
+    interlock_ids = (interlock_ids - params[:entity_ids].split(',').map(&:to_i)).take(num)
+
+    if interlock_ids.count > 0
```

Comment:
I suggest that this code be moved from the controller. It could go in a Service object or a method on Oligrapher or Network Map.